src/components/data-table/ha-data-table.ts

```diff
@@ -144,48 +155,22 @@ export class HaDataTable extends BaseElement {
       direction: SortingDirection,
       sortColumn: string
     ) => {
-      const sorted = [...data];
-      const column = columns[sortColumn];
-      return sorted.sort((a, b) => {
-        let sort = 1;
-        if (direction === "desc") {
-          sort = -1;
-        }
-
-        let valA = column.filterKey
-          ? a[sortColumn][column.filterKey]
-          : a[sortColumn];
-
-        let valB = column.filterKey
-          ? b[sortColumn][column.filterKey]
-          : b[sortColumn];
-
-        if (typeof valA === "string") {
-          valA = valA.toUpperCase();
-        }
-        if (typeof valB === "string") {
-          valB = valB.toUpperCase();
-        }
-
-        if (valA < valB) {
-          return sort * -1;
-        }
-        if (valA > valB) {
-          return sort * 1;
-        }
-        return 0;
-      });
+      const column = { ...columns[sortColumn] };
+      delete column.template;
+      return worker.sortData(data, column, direction, sortColumn);
```

Comment:
Every time we send data to the worker, we need to copy it all over to the worker. Could we do filtering and sorting in one go ? Basically we get a "slim" column definition just for sorting that we can send over.